Jenkinsfile

```diff
@@ -88,93 +88,237 @@ try {
+                (compiler == "clang") ? '/opt/llvm-5.0.1/bin/clang' : 'gcc'
+            def cxx =
+                (compiler == "clang") ? '/opt/llvm-5.0.1/bin/clang++' : 'g++'
+            def ucc = isNoUnity(target) ? 'true' : 'false'
+            def node_type =
+                (compiler == 'msvc') ? 'rippled-win' : 'rippled-dev'
+
+            def env_vars = [
+                "TARGET=${target}",
+                "COMPILER=${compiler}",
+                "PARALLEL_TESTS=true",
+                'BUILD=cmake',
+                "BUILD_DIR=${bldlabel}",
+                "CMAKE_EXTRA_ARGS=${cmake_extra}",
+                'VERBOSE_BUILD=true']
+
+            builds[bldlabel] = {
+                node(node_type) {
                     checkout scm
                     dir ('build') {
                         deleteDir()
                     }
                     def cdir = upDir(pwd())
                     echo "BASEDIR: ${cdir}"
-                    def compiler = getCompiler(bldtype)
-                    def target = getTarget(bldtype)
-                    if (compiler == "coverage") {
-                        compiler = 'gcc'
-                    }
                     echo "COMPILER: ${compiler}"
                     echo "TARGET: ${target}"
-                    def clang_cc  =
-                        (compiler == "clang") ? "${LLVM_ROOT}/bin/clang" : ''
-                    def clang_cxx =
-                        (compiler == "clang") ? "${LLVM_ROOT}/bin/clang++" : ''
-                    def ucc = isNoUnity(target) ? 'true' : 'false'
                     echo "USE_CC: ${ucc}"
-                    withEnv(["CCACHE_BASEDIR=${cdir}",
-                             "CCACHE_NOHASHDIR=true",
-                             'LCOV_ROOT=""',
-                             'BOOST_ROOT=/opt/local/boost_1_66_0',
-                             "TARGET=${target}",
-                             "CC=${compiler}",
-                             'BUILD=cmake',
-                             'VERBOSE_BUILD=true',
-                             "CLANG_CC=${clang_cc}",
-                             "CLANG_CXX=${clang_cxx}",
-                             "USE_CCACHE=${ucc}"])
+
+                    if (compiler == 'msvc') {
+                        env_vars.addAll([
+                            "BOOST_ROOT=c:\\lib\\boost_1_66",
+                            'MSBUILDDISABLENODEREUSE=1',
+                            "OPENSSL_ROOT=c:\\OpenSSL-Win64"])
+                    }
+                    else {
+                        env_vars.addAll([
+                            "NINJA_BUILD=false",
+                            "CCACHE_BASEDIR=${cdir}",
+                            'PLANTUML_JAR=/opt/plantuml/plantuml.jar',
+                            "CCACHE_NOHASHDIR=true",
+                            "CC=${cc}",
+                            "CXX=${cxx}",
+                            'LCOV_ROOT=""',
+                            "PATH+CMAKE_BIN=/opt/local/cmake",
```

Comment:
Is `PATH+CMAKE_BIN` a special syntax?

Reply:
yep - prepending:

https://jenkins.io/doc/pipeline/steps/workflow-basic-steps/#code-withenv-code-set-environment-variables